app/main.py

```diff
@@ -1,25 +1,64 @@
 from abc import ABC
+from typing import Callable
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self,
+                instance: object,
+                owner: Callable) -> None:
+        return getattr(instance, self.protected_name)
```

Comment:
will this method return None ?

Reply:
It returns value, I have already changed None for int.